pkg/scheduler/framework/plugins/podtopologyspread/plugin.go

```diff
@@ -141,9 +145,126 @@ func (pl *PodTopologySpread) EventsToRegister() []framework.ClusterEventWithHint
+		logger.V(5).Info("a scheduled pod was created, but it doesn't matches with the pod's topology spread", "pod", klog.KObj(pod), "createdPod", klog.KObj(originalPod))
+		return framework.QueueSkip, nil
+	}
+
+	// Pod is deleted. Return Queue when the deleted Pod has a label that matches with topologySpread's selector.
+	if len(modifiedPod.Spec.NodeName) == 0 {
+		logger.V(5).Info("the deleted pod is unscheduled and it doesn't make the target pod schedulable", "pod", klog.KObj(pod), "originalPod", klog.KObj(modifiedPod))
+		return framework.QueueSkip, nil
+	}
+	if podLabelsMatchSpreadConstraints(constraints, modifiedPod.Labels) {
+		logger.V(5).Info("a scheduled pod which matches with the pod's topology spread was deleted, and the pod may be schedulable now", "pod", klog.KObj(pod), "deletedPod", klog.KObj(modifiedPod))
+		return framework.Queue, nil
+	}
+	logger.V(5).Info("a scheduled pod was deleted, but it's unrelated to the pod's topology spread", "pod", klog.KObj(pod), "deletedPod", klog.KObj(modifiedPod))
+
+	return framework.QueueSkip, nil
+}
+
+// getConstraints extracts topologySpreadConstraint(s) from the Pod spec.
+// If the Pod doesn't have any topologySpreadConstraint, it returns default constraints.
+func (pl *PodTopologySpread) getConstraints(pod *v1.Pod) ([]topologySpreadConstraint, error) {
+	var constraints []topologySpreadConstraint
+	var err error
+	if len(pod.Spec.TopologySpreadConstraints) > 0 {
+		// We have feature gating in APIServer to strip the spec
+		// so don't need to re-check feature gate, just check length of Constraints.
+		constraints, err = pl.filterTopologySpreadConstraints(
+			pod.Spec.TopologySpreadConstraints,
+			pod.Labels,
+			v1.DoNotSchedule,
+		)
+		if err != nil {
+			return nil, fmt.Errorf("obtaining pod's hard topology spread constraints: %w", err)
+		}
+	} else {
+		constraints, err = pl.buildDefaultConstraints(pod, v1.DoNotSchedule)
+		if err != nil {
+			return nil, fmt.Errorf("setting default hard topology spread constraints: %w", err)
+		}
+	}
+	return constraints, nil
+}
+
+func (pl *PodTopologySpread) isSchedulableAfterNodeChange(logger klog.Logger, pod *v1.Pod, oldObj, newObj interface{}) (framework.QueueingHint, error) {
+	originalNode, modifiedNode, err := util.As[*v1.Node](oldObj, newObj)
+	if err != nil {
+		logger.Error(err, "unexpected objects in isSchedulableAfterNodeChange", "oldObj", oldObj, "newObj", newObj)
+		return framework.Queue, err
+	}
+
+	constraints, err := pl.getConstraints(pod)
+	if err != nil {
+		logger.Error(err, "get constraints from pod failed", "pod", klog.KObj(pod))
+		return framework.Queue, err
+	}
+
+	// framework.Add/framework.Update: return Queue when node has topologyKey in its labels, else return QueueSkip.
+	if modifiedNode != nil {
+		if !nodeLabelsMatchSpreadConstraints(modifiedNode.Labels, constraints) {
+			logger.V(5).Info("the created/updated node doesn't match pod topologySpreadConstraints", "pod", klog.KObj(pod), "node", klog.KObj(modifiedNode))
```

Comment:
Now, I'd like to unify how we call pod topology spread in logs and errors, there are three now:
- topologySpreadConstraints
- topology spread
- topology spread constraints

I think the last one "topology spread constraints" is the clearest. Can you unify in all logs and errors implemented in this PR?